music21/stream/__init__.py

```diff
@@ -2709,6 +2709,105 @@ def replaceDerived(startSite=self):
 
         replaceDerived()
 
+    def splitAtDurations(self, *, recurse=False) -> Optional[Dict]:
```

Comment:
I'm trying to keep the return signatures and the non-keyword arguments of overridden methods as close to the original as possible.  So my preference would be that the return signature is the same as the original -- even though it's a bit ridiculous.  For one: Streams are not really hashable (I'm surprised that it allowed it as the key for a dict) -- even if we keep it, it should return an empty dict if nothing is found.  There's something that can bite you about:

```
# simplify durations:
for el in s.recurse():
     didSplit = el.splitAtDurations()
     if didSplit:
         print(didSplit[0])  # Bang! dies on a nested stream.
```

IDEs and typecheckers won't know why el.splitAtDurations() returned a Dict this time instead of a List.

Maybe we could have a `returnChangedElements=False` keyword where if someone specifically opts in to this, it returns something more useful.  But that should be List[Tuple[Stream, List[Music21Object]]] to avoid the hashing on Streams.  So then you can do:

```
for changedStream, changedElements in s.splitAtDurations(recurse=True, returnChangedElements=True):
     print(f'{changedStream}: {changedElements}')
```

Reply:
Agree. Thanks for sketching out the opt-in keyword, but I don't think we need the complexity. I like just returning the sort tuple. There would be other ways to track containers that need to have tuplets re-drawn, like setting a stream status flag in the subroutine while it's being edited.

My question is -- do you still want the stream itself to have its durations split? So a 5.0QL stream would be split into a 4.0QL stream and a 1.0QL stream? I'm thinking that's unnecessary.

Reply:
Ah, right, it's moot because it wasn't working (AttributeError: 'tuple' object has no attribute 'spannerList').